packages/fetch/src/types.ts

```diff
@@ -16,182 +16,177 @@
    */
   releaseLock(): void;
-} & AsyncIterable<Uint8Array>;
-
-// Environment independent Fetch API
-
-type FetchRequestInfo = FetchRequest | string;
-type FetchHeadersInit = FetchHeaders | string[][] | Record<string, string>;
-type FetchRequestCredentials = "include" | "omit" | "same-origin";
-type FetchRequestMode = "cors" | "navigate" | "no-cors" | "same-origin";
-
-interface FetchBody {
-  readonly body: ReadableStream | null;
-  readonly bodyUsed: boolean;
-  arrayBuffer(): Promise<ArrayBuffer>;
-  blob(): Promise<unknown>;
-  json<ResponseBody = unknown>(): Promise<ResponseBody>;
-  text(): Promise<string>;
-}
-
-export interface FetchHeaders {
-  append(name: string, value: string): void;
-  delete(name: string): void;
-  get(name: string): string | null;
-  has(name: string): boolean;
-  set(name: string, value: string): void;
-  forEach(callback: (value: string, name: string) => void): void;
-}
-
-/** This Fetch API interface represents a resource request. */
-export interface FetchRequest extends FetchBody {
-  /**
-   * Returns a Headers object consisting of the headers associated with request. Note that headers added in the network layer by the user agent will not be accounted for in this object, e.g., the "Host" header.
-   */
-  readonly headers: FetchHeaders;
-  /**
-   * Returns a boolean indicating whether or not request can outlive the global in which it was created.
-   */
-  readonly keepalive: boolean;
-  /**
-   * Returns request's HTTP method, which is "GET" by default.
-   */
-  readonly method: string;
-  /**
-   * Returns the signal associated with request, which is an AbortSignal object indicating whether or not request has been aborted, and its abort event handler.
-   */
-  readonly signal: AbortSignal;
-  /**
-   * Returns the URL of request as a string.
-   */
-  readonly url: string;
-  clone(): FetchRequest;
-}
-
-export interface FetchResponse extends FetchBody {
-  readonly headers: FetchHeaders;
-  readonly ok: boolean;
-  readonly redirected: boolean;
-  readonly status: number;
-  readonly statusText: string;
-  readonly type: unknown;
-  readonly url: string;
-  clone(): FetchResponse;
-}
-
-export interface FetchRequestInit<RequestBody = unknown> {
+}; //  & AsyncIterable<Uint8Array>;
+
+export declare function fetch<
+  PlatformRequestBody = RequestBody,
+  PlatformHeaders extends Headers = Headers,
+  PlatformAbortSignal extends AbortSignal = AbortSignal,
+  PlatformResponseBody = ReadableStream | null,
+>(
+  input: string,
+  init?: RequestInit<PlatformRequestBody, PlatformHeaders, PlatformAbortSignal>,
+): Promise<Response<PlatformHeaders, PlatformResponseBody>>;
+
+/* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+export type AnyFetch = typeof fetch<any, any, any, any>;
+
+export interface RequestInit<
+  PlatformRequestBody = RequestBody,
+  PlatformHeaders = Headers,
+  PlatformAbortSignal = AbortSignal,
+> {
   /**
    * A BodyInit object or null to set request's body.
```

Comment:
The docs of these properties describe the allowed types, but after these updates some seem to have changed. Describing the types (e.g. `a string`, `an object`) in the docs (as compared to only via the TS types) could make the docs and the actual allowed types out of sync. E.g. for this one, I believe also `ReadableStream` and `string` are allowed.
```suggestion
   * The request's body.
```